src/Compiler/SyntaxTree/SyntaxTree.fs

```diff
@@ -308,21 +308,18 @@ type SeqExprOnly = SeqExprOnly of bool
 
 [<NoEquality; NoComparison; RequireQualifiedAccess>]
 type BlockSeparator =
-    | Semicolon of range: range * position: pos option
-    | Comma of range: range * position: pos option
-    | Offside of range: range * position: pos option
+    | Semicolon of range: range * position: pos
+    | Comma of range: range * position: pos
```

Comment:
Given that the both range and position are non-optional and pos is always at the end of the range, do we actually need to keep both?

Reply:
Ahh I see. Not I do not think we need both.